library/core/src/slice/sort.rs

```diff
@@ -101,15 +101,21 @@ where
             // into the slice.
             let mut tmp = mem::ManuallyDrop::new(ptr::read(v.get_unchecked(len - 1)));
             let mut hole = CopyOnDrop { src: &mut *tmp, dest: v.get_unchecked_mut(len - 2) };
-            ptr::copy_nonoverlapping(v.get_unchecked(len - 2), v.get_unchecked_mut(len - 1), 1);
+            let ptr = v.as_mut_ptr();
+            let src = (ptr as *const T).offset(len as isize - 2);
```

Comment:
nit: `.add(len - 2)` seems better here, we don't want negative offsets here I assume?